src/ui/components/Scanner/Scanner.tsx

```diff
@@ -86,8 +89,11 @@ const Scanner = forwardRef(
     };
 
     const stopScan = async () => {
+      if (permission) {
```

Comment:
Wondering do we actually need permission to stop scanning? In theory if we had no permission we wouldn't be scanning, right?

Reply:
on unmount it'll stop scanning to clean up. So _trying_ to stop scanning even when not already scanning because of permission issues could cause a problem

Reply:
Because this func called in `processValue` and you know `processValue` also called when user submit value that manual input on input modal. So we need check permission here and prevent `stopScan` function.

Reply:
Maybe we can take `stopScan` out of `processValue` and call `stopScan` only before `processValue` when the user has scanned something and not when a value was entered manually.

Reply:
We should have moved that out when we introduced the manual input.

Reply:
If user allow camera permission but they don't use camera to scan and they input value manually. In this case we still need call stop scan to hide camera screen.

Reply:
Approving for now since this will go to your feature branch anyway, so we can free Duke from this one. Feel free to change things anyway if you feel like this is wrong (I know you're busy at the moment 🚣 )

Reply:
@sdisalvo-crd imo this is the correct way to do it.. `stopScan` is also called when the component is unmounting. It's much safer to do this